### PR TITLE
Update dependencies, glob, and IntelliSense loc strings

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Some dependencies were not updated due to breaking changes. We could follow up on those later.

In particular, glob was updated since the older version was giving warnings about it not being maintained anymore.

FYI, I encountered a bug with yarn 1.22.10 that was fixed with 1.22.22 in case anyone else hits it when trying to regenerate the yarn.lock.

### CODE REVIEW

File: Extension/src/LanguageServer/codeAnalysis.ts [hunk 1]
```diff
@@ -378,7 +378,7 @@ export function publishCodeAnalysisDiagnostics(params: PublishCodeAnalysisDiagno
                     docPage = `checks${checksGroup}/${checksPage}.html`;
                 }
                 // TODO: This should be checking the clang-tidy version used to better support usage of older versions.
-                const primaryDocUri: vscode.Uri = vscode.Uri.parse(`https://releases.llvm.org/18.1.1/tools/clang/tools/extra/docs/clang-tidy/${docPage}`);
+                const primaryDocUri: vscode.Uri = vscode.Uri.parse(`https://releases.llvm.org/18.1.4/tools/clang/tools/extra/docs/clang-tidy/${docPage}`);
```
Comment (bobbrow) [inline Extension/src/LanguageServer/codeAnalysis.ts]: should it be 18.1.5?

**Comment by sean-mcmanus** [comment]
I can't update ThirdPartyNotices.txt for some unknown reason...

**Comment by sean-mcmanus** [comment]
> I can't update ThirdPartyNotices.txt for some unknown reason...

It's updated now.

**Review by Colengms — DISMISSED** [review]

**Comment by sean-mcmanus** [comment]
I'll re-update this since it's been 3 weeks.

**Review by Colengms — APPROVED** [review]

**Pull request merged**